### PR TITLE
Closes #2218 Added `DTHCAUS`, `DTHCGR1` to ADSL template and vignette

### LINKED ISSUE #2218: Add DTHCGR1 and DTHCAUS to ADSL template
### Background Information

Please add death variables DTHCAT and DTHCAUS to the admiral ADSL templates, this is to support the creation of death outputs using {pharmaverseadam} in the TLG-catalog.

### Definition of Done

see above

### PR DESCRIPTION
**Pull request by manciniedoardo**
Closes #2218 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@khatril FYI - this will soon be merged into admiral and then we just need to do a re-run on pharmaverseadam to reflect those changes in the package. From your side it will just be necessary to use DTHCGR1 instead of DTHCAT in any TLG code.

File: NEWS.md
Comment (bms63) [inline NEWS.md:145]: Could we just put this in the Various section and not make a new section in our changelog and maintain consistency??

File: inst/templates/ad_adsl.R
Comment (bms63) [inline inst/templates/ad_adsl.R:217]: Do we want to use the new functions that @jeffreyad just made in https://github.com/pharmaverse/admiral/pull/2236 since we are superseding this function?

File: NEWS.md
Comment (manciniedoardo) [reply NEWS.md]: Thanks, actioned.

File: inst/templates/ad_adsl.R
Comment (manciniedoardo) [reply inst/templates/ad_adsl.R:217]: Really good point, thanks - it had slipped my mind. I will thus wait until #2236 is merged and then use that instead.

Comment (jeffreyad) [reply inst/templates/ad_adsl.R:217]: Here is some example code for using `derive_vars_extreme_event()`

```
adsl <- adsl %>%
  derive_vars_extreme_event(
    by_vars = exprs(STUDYID, USUBJID),
    events = list(
      event(
        dataset_name = "ae",
        condition = AEOUT == "FATAL",
        set_values_to = exprs(DTHCAUS = AEDECOD),
      ),
      event(
        dataset_name = "ds",
        condition = DSDECOD == "DEATH" & grepl("DEATH DUE TO", DSTERM),
        set_values_to = exprs(DTHCAUS = DSTERM),
      )
    ),
    source_datasets = list(ae = ae, ds = ds),
    tmp_event_nr_var = event_nr,
    order = exprs(event_nr),
    mode = "first",
    new_vars = exprs(DTHCAUS = DTHCAUS)
  )
```

File: vignettes/adsl.Rmd
Comment (bms63) [inline vignettes/adsl.Rmd:1]: Should we display some data with death Dates here?  Currently it is all `NAs`

image

**Review by bms63 — APPROVED** [review]
Just a minor suggestion.  Looks good!

File: vignettes/adsl.Rmd
Comment (manciniedoardo) [reply vignettes/adsl.Rmd:1]: (let me know if it's fine to modify the vignette in this way @bms63)

File: vignettes/adsl.Rmd
Comment (bms63) [inline vignettes/adsl.Rmd:393]: so fancy

**Review by bms63 — APPROVED** [review]

**Pull request merged**
